// filepath: /c:/chessApp/cloudtaak_chessapp/chess-app/soap-adapter/adaptor/proto/games.proto
syntax = "proto3";

package games;

service GameService {
  rpc GetGamesByPlayer (PlayerRequest) returns (GameListResponse);
  rpc GetMoveSuggestion (MoveRequest) returns (MoveResponse);
}

message PlayerRequest {
  string player = 1;
}

message MoveRequest {
  int32 game_id = 1;
}

message Game {
  int32 id = 1;
  string black = 2;
  string white = 3;
  string winner = 4;
  string moves = 5;
  string created_at = 6;
}

message GameListResponse {
  repeated Game games = 1;
}

message MoveResponse {
  string suggestion = 1;
}